gramineproject/gramine
# Rework sgx_include_path and build against linux-libc-dev

## How to test this PR? 

On Ubuntu: try 20.04 (should build from aux kernel sources, as it has kernel < 5.11) and 22.04 (has upstream driver).

If you have only a single machine/containser: use Debian 11 and  you need to either have installed or not installed `linux-libc-dev (>= 5.11)`.

```sh
apt-get install kernel-headers-amd64 -t bullseye-backports
apt-get build-dep . -t bullseye-backports # shouldn't install newer linux-libc-dev, but check it
debuild # should build with aux kernel
apt-get install linux-libc-dev -t bullseye-backports
debuild # should build with #include <asm/sgx.h>
```

CI will build only `#include <asm/sgx.h>`.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/1090)
